### ISSUE TITLE
How to get choice from LayerChoice/InputChoice during search?

### ISSUE THREAD

**Issue by kingaza** [microsoft/nni#2667]
For example in nas/naive
`self.conv1 = LayerChoice([nn.Conv2d(3, 6, 3, padding=1), nn.Conv2d(3, 6, 5, padding=2)])`
I want to move the padding from `__init__()` to function `forward()`
But I don't know which choice has been made in current step.
Any support is appreciated.

**Comment by ultmaster** [comment]
In current NNI design, if you set `return_mask=True`, you will get a choice vector along with the output tensor as the result of calling `self.conv1`. Please check whether this satisfies your need.

However, there is no way you can know which choice has been made before you even call it. If you are interested in getting a choice beforehand, go into the details to customize mutator -- think of yourself as a NAS developer.

If your case is as simple as you just described, you can totally infer padding in `forward()` of your module. In this way, you can still treat all choices equally.

**Comment by scarlett2018** [comment]
@kingaza - thanks for inquiring us for questions, pls help to close the issue if @ultmaster 's answer works for you. feel free to continue the discussion if the answer is not yet sufficient.

**Comment by kingaza** [comment]
So for late replay. @scarlett2018 , and thanks @ultmaster  for your help.
I am still trying to figure out if it is a real requirement or not
This requirement might make forward() to handle too many things. I am wondering if there is a more elegant solution.

**Comment by scarlett2018** [comment]
close the issue as there are no further inquiry or discussion for long time.
